src/main/java/io/appium/java_client/remote/MobileOptions.java

```diff
@@ -0,0 +1,515 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.java_client.remote;
+
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.MutableCapabilities;
+import org.openqa.selenium.ScreenOrientation;
+import org.openqa.selenium.remote.CapabilityType;
+
+import java.net.URL;
+import java.time.Duration;
+
+public class MobileOptions<T extends MobileOptions<T>> extends MutableCapabilities {
+
+    /**
+     * Creates new instance with no preset capabilities.
+     */
+    public MobileOptions() {
+    }
+
+    /**
+     * Creates new instance with provided capabilities capabilities.
+     *
+     * @param source is Capabilities instance to merge into new instance
+     */
+    public MobileOptions(Capabilities source) {
+        merge(source);
+    }
+
+    /**
+     * Set the kind of mobile device or emulator to use.
+     *
+     * @param platform the kind of mobile device or emulator to use.
+     * @return this MobileOptions, for chaining.
+     * @see org.openqa.selenium.remote.CapabilityType#PLATFORM_NAME
+     */
+    public T setPlatformName(String platform) {
```

Comment:
this option is not just mobile, but is also applicable to other driver, like for example the Mac OS or Windows one. I was also thinking about what we could do with these. Perhaps, it makes sense to extract such common options into an interface with default method implementations (aka java-like mixin) and add this interface into concrete option classes for different platforms?

Reply:
Yeah, but of the 10 capabilities specified in the w3c, this is really the only one that applies to mobile, so I don't think it is worth spending time on the overlap.